packages/clerk-js/src/utils/componentGuards.ts

```diff
@@ -7,7 +7,7 @@ export type ComponentGuard = (
 ) => boolean;
 
 export const sessionExistsAndSingleSessionModeEnabled: ComponentGuard = (clerk, environment) => {
-  return !!(clerk.session && environment?.authConfig.singleSessionMode);
+  return !!(clerk.session?.status === 'active' && environment?.authConfig.singleSessionMode);
```

Comment:
```suggestion
  return !!(clerk.hasValidSession && environment?.authConfig.singleSessionMode);
```

This will avoid breaking changes for statuses changes